### ISSUE TITLE
AmazonMQ socket is closed continously

### ISSUE THREAD

**Issue by jcampbell05** [celery/celery#5001]
## Checklist

- [x] I have included the output of ``celery -A proj report`` in the issue.
      (if you are not able to do this, then at least specify the Celery
       version affected).
- [x] I have verified that the issue exists against the `master` branch of Celery.

Version 4.2, Python 2.7.14

## Steps to reproduce

Spin up a AmaoznMQ server and try to connect to it. 

## Expected behavior

Celery should be able to connect to amazonMQ over AMQP

## Actual behavior

See that celery spits out socket closed continously.

**Comment by clokep** [comment]
I think [Amazon MQ](https://aws.amazon.com/amazon-mq/) only supports AMQP 1.0 (since it is based on ActiveMQ under the hood) (this [Migrating from RabbitMQ to Amazon MQ](https://aws.amazon.com/blogs/compute/migrating-from-rabbitmq-to-amazon-mq/) article seems to mention AMQP 1.0 specifically many times for Amazon MQ.

I don't think any of the kombu brokers support AMQP 1.0 at the moment.

**Comment by jcampbell05** [comment]
@clokep do we know if there is an issue tracking AMQP 1.0 support? I've read in a previous issue that this was supposed to be the case in Celery 4.X so some clarity would be awesome :)

**Comment by clokep** [comment]
@jcampbell05 celery/kombu#548 and celery/kombu#723 are the vaguely related ones I know about.

**Comment by jcampbell05** [comment]
So I've been able to find out the only transport method which Kombu supports which enables this is QPID, but that fails with this error when I run it:
`Transport' object has no attribute 'use_async_interface'`

**Comment by yardensachs** [comment]
Is there plans to support ActiveMQ / AmazonMQ?

**Comment by auvipy** [comment]
yes in celery 5. feel free to donate for the effort.

**Comment by auvipy** [comment]
https://www.rabbitmq.com/interoperability.html

**Comment by jheld** [comment]
@auvipy in general, for funding or feature development, could you elaborate what is needed to make this a reasonable feature? How much funding, or what items technically need to be researched/completed? I do see the interop link (thank you), but do you have anything specific in mind?

**Comment by auvipy** [comment]
If you are serious I would like to take the time to come up with a reasonable time and works needed for all these

**Comment by jheld** [comment]
Yes. I'd be interested to provide the cost analytics to my DevOps team and see what they think (which I can get from the AWS product docs). Redis is fine (quite good actually), but ActiveMQ/AWS might be better, and let's them manage it.

I'll check with my team (hopefully this week), and see what they think. In general, I'm serious anyway!

**Comment by auvipy** [comment]
> Are there any plans to support ActiveMQ / AmazonMQ?

@yardensachs any plan to donate?

**Comment by auvipy** [comment]
> I think [Amazon MQ](https://aws.amazon.com/amazon-mq/) only supports AMQP 1.0 (since it is based on ActiveMQ under the hood) (this [Migrating from RabbitMQ to Amazon MQ](https://aws.amazon.com/blogs/compute/migrating-from-rabbitmq-to-amazon-mq/) article seems to mention AMQP 1.0 specifically many times for Amazon MQ.
> 
> I don't think any of the kombu brokers support AMQP 1.0 at the moment.

@clokep you are right. we wil add amqp 1.0 to most of the brokers which are supported

**Comment by yardensachs** [comment]
@auvipy donate time you mean? or do you mean funding?

**Comment by auvipy** [comment]
> @auvipy donate time you mean? or do you mean funding?

financial donation :)

**Comment by yardensachs** [comment]
I am sorry, not at the moment. I am happy to help with coding a solution @auvipy

**Comment by jheld** [comment]
@auvipy could/should we consider the amqp 1.0 support something that belongs as a CEP?

**Comment by auvipy** [comment]
> @auvipy could/should we consider the amqp 1.0 support something that belongs as a CEP?

yes we should

**Comment by dejlek** [comment]
I would also be interested in helping with this as I am looking forward to be able to replace ElastiCache with AWS AMQ as a broker...

**Comment by auvipy** [comment]
I will start working on amqp-1.0 support very soon. targeting celery 4.5. financial contributions are hugely welcome.

**Comment by jheld** [comment]
My company is looking to use AMQ possibly in short/medium term.

I've let my supervisor know that not only is this a feature where work is
being done, but that financial contributions would greatly help.

I'll update as I hear.

**Comment by jheld** [comment]
@auvipy forgive me I've forgotten, what is the proposed timeline for 4.5 beta and release?

**Comment by auvipy** [comment]
I would like to introduce amqp-1.0 and updated billiard 4.0 by august 2020 and possibly the first rc1 by mid-september. I will resume those efforts after the celery 4.4.3 release this week.

**Comment by jheld** [comment]
The business driver of what would have pushed for this upgrade has been moved to later this year/next year.

We are still interested in it (I've had it as a tentative road map item since May of last year, unrelated to any business decisions).

Should anything change (closer timeline and thus possible finance support), I'll update here/appropriate channel.

**Comment by auvipy** [comment]
this will be python 3 only ;)

**Comment by jheld** [comment]
So it's slated for celery 5?

**Comment by auvipy** [comment]
no. celery 4.4 will be last to support python 2.7. 4.5+ will be celery 3.5+ only. and for practical reason

**Comment by auvipy** [comment]
celery 5 is a long overdue and will take longer than anticipated. so I will start introducing non-breaking new features to celery 4.5 onwards

**Comment by jheld** [comment]
I realize this isn't the proper place for a discussion on this but is there a policy doc for which features we're willing to backport to the 4.4.x releases since everything onward has the ability not to support py2.7 compatibility. I agree that moving to py3 ahead of celery5 is generally a good move, but we need to be prepared for folks to clamor for certain changes (and be told they'll have to do it manually!) and not be super happy with our answer.

**Comment by auvipy** [comment]
4.4.x will get the bug fix backports contributed by the community. and also need to have some funds to support time. my intension is 4.4.x to be supported till 30 december 2020. minor bug fix backports can be easily released.

**Comment by romange** [comment]
@dejlek  Could you share why you want to migrate from ElastiCache to AMQ?

**Comment by legshort** [comment]
AWS Active MQ now supports Rabbit MQ 3.8.6 🎉🎉
https://aws.amazon.com/ko/about-aws/whats-new/2020/11/announcing-amazon-mq-rabbitmq/
